### PR TITLE
Issue 2038 fix flaky tests

### LINKED ISSUE #2038: Fix flaky tests
I've temporarily excluded flaky tests. They will be fixed in a separate issue.

_Originally posted by @DenBond7 in https://github.com/FlowCrypt/flowcrypt-android/issues/2013#issuecomment-1293111618_

### PR DESCRIPTION
**Pull request by DenBond7**
This PR fixed some flaky tests

close #2038

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/junit/filters/BaseCustomFilter.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        && description.getAnnotation(annotationClass) == null
+  }
+
+  protected fun isAnnotationPresentAtClassOrMethod(
+    description: Description?,
+    annotationClass: Class<out Annotation?>
+  ): Boolean {
+    return description?.testClass?.isAnnotationPresent(annotationClass) == true
+        || description?.getAnnotation(annotationClass) != null
+  }
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/junit/filters/BaseCustomFilter.kt:32]: I think we only need one `isAnnotationPresentAtClassOrMethod` method.
And `isAnnotationNotPresentAtClassNorMethod()` can be called as `!isAnnotationPresentAtClassOrMethod()`

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/junit/filters/BaseCustomFilter.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  protected fun isAnnotationNotPresentAtClassOrMethod(
+    description: Description?,
+    annotationClass: Class<out Annotation?>,
+    parentFunction: (Description?) -> Boolean
+  ): Boolean {
+    val hasClassAnnotation = description?.testClass?.isAnnotationPresent(annotationClass) == true
+    if (hasClassAnnotation) return false
+    return parentFunction(description) && description?.getAnnotation(annotationClass) == null
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/junit/filters/BaseCustomFilter.kt:41]: Is this function used anywhere?

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/MainSignInFragmentEnterpriseFlowTest.kt [hunk 3]
```diff
@@ -72,23 +76,38 @@ class MainSignInFragmentEnterpriseFlowTest : BaseSignTest() {
...
-            return handleFesAvailabilityAPI(gson)
+        when {
+          request.path?.startsWith("/api") == true -> {
+            if (request.path.equals("/api/")) {
+              return handleFesAvailabilityAPI(gson)
+            }
+
+            if (request.path.equals("/api/v1/client-configuration?domain=localhost:1212")) {
+              return handleClientConfigurationAPI(gson)
+            }
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/MainSignInFragmentEnterpriseFlowTest.kt:87]: we don't need `if (request.path?.startsWith("/api") == true) {` and can be written as below

```kt
request.path.equals("/api/") -> {
  return handleFesAvailabilityAPI(gson)
}
request.path.equals("/api/v1/client-configuration?domain=localhost:1212") -> {
  return handleClientConfigurationAPI(gson)
}
```

**Review by ioanmo226 — COMMENTED** [review]
Looks good 👍 
Please see below comments

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/junit/filters/BaseCustomFilter.kt (hunk 2 above)
Comment (DenBond7) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/junit/filters/BaseCustomFilter.kt:41]: ha... It is not used after some code refactoring. Unfortunately, IDEA didn't show me a hint. Good notice!

**Review by ioanmo226 — APPROVED** [review]
👍

**Pull request merged**